### ISSUE TITLE
[silicon_creator] implement ECDSA sigverify operations

### ISSUE THREAD

**Issue by timothytrippel** [lowRISC/opentitan#21548]
Implement the ECDSA-equivalent sigverify operations on OTBN:
- [x] OTBN (#21824)
- [ ] https://github.com/lowRISC/opentitan/pull/23680
- Ibex (dropped)

Additionally, integrate these into the sigverify silicon_creator device library.

**Comment by moidx** [comment]
Hi @jadephilipoom, @cfrantz, @timothytrippel, @vogelpi, @andreaskurth 

Currently we have been using OTBN execution in both silicon and FPGA environments in `earlgrey_es_sival` and `master` branches. 

`top_earlgrey_es` has an OTP configuration word that supports reverting to a software implementation of RSA in case there were hardware issues with OTBN during silicon bring-up. The option was not used in SiVal, and all our testing has thus far been based on OTBN performing signature verification.

Given that OTBN has so far been proven to work, I propose we drop the request to have support for ECDSA in Ibex. This is mainly due to the following reasons: 

1. Code size space constraints. We are currently with about 1KB of room left in the ROM, and we still have to implement some fixes for `Earlgrey-PROD.ROM`.
2. We can explore the option of running SLH-DSA and ECDSA in parallel to help optimize boot performance.

The 2. point above does not need to be part of `Earlgrey-PROD.ROM`, but can potentially be considered for `Earlgrey-PROD.ROM_EXT`. 

To further mitigate risk of not having an ECDSA software backup, we have focused on re-enabling ROM e2e test cases in DV. We will run a sub set of these tests in gate level simulation to include them to the tape-out sign-off criteria.

Please add your comments here and feel free to add this topic to any relevant working group agendas for further discussion.

**Comment by jadephilipoom** [comment]
This sounds okay to me. It's nice to have redundancy in ROM, but the SPHINCS+ implementation gives us a little bit of protection already. If there was a security-critical issue with the ECDSA implementation that meant it accepted invalid signatures (or was vulnerable e.g. to a fault injection attack that made it accept invalid signatures), then we could always enable SPHINCS+ as an emergency mitigation. If the ECDSA implementation rejected valid signatures, then we wouldn't have an alternative, but this scenario is much less dangerous and seems unlikely to occur in a way that would affect large numbers devices/keys. For example, if there was a fault-injection attack that made the ECDSA implementation reject a valid signature, that wouldn't really be a big problem; it would just make the device fail to boot.

So the risks seem acceptable to me, and now that we're getting down to the wire in terms of ROM space and development time I think it makes sense to accept it in order to narrow the scope here.
